[jit] Split ClassType into a separate header.

zhxchen17 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69483
* #68137
* #69482
* #68136
* #68038
* #68037
* __->__ #68036



In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D32264618/)!

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618)

zhxchen17 committed:
[jit] Split ClassType into a separate header.

In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

[ghstack-poisoned]

iseeyuan commented:
> in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Are you sure only class_type is needed for Edge? How could we work around container types like Dict?

zhxchen17 commented:
> > in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h
> 
> Are you sure only class_type is needed for Edge? How could we work around container types like Dict?

Yeah my major point is to provide a separate implementation for types like Dict, which is an unpublished internal diff right now. These diffs are just preparations for that diff. I will send in soon.

tugsbayasgalan reviewed (approved):
Looks good to me! What if we also move cpp logic of classtype to separate file while we are at it?

zhxchen17 committed:
Update on "[jit] Split ClassType into a separate header."


In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Split ClassType into a separate header."


In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Split ClassType into a separate header."


In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Split ClassType into a separate header."




In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D32264618/)!

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Split ClassType into a separate header."




In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D32264618/)!

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Split ClassType into a separate header."




In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D32264618/)!

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Split ClassType into a separate header."




In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D32264618/)!

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Split ClassType into a separate header."




In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D32264618/)!

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Split ClassType into a separate header."




In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D32264618/)!

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Split ClassType into a separate header."




In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D32264618/)!

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Split ClassType into a separate header."




In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D32264618/)!

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Split ClassType into a separate header."




In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D32264618/)!

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Split ClassType into a separate header."




In Edge cases we want to separately include class_type.h because in the future we want to stop depending on the rest of the JIT types declared inside jit_type.h

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D32264618/)!

Differential Revision: [D32264618](https://our.internmc.facebook.com/intern/diff/D32264618)

[ghstack-poisoned]
